### ISSUE TITLE
Cannot reproduce the results in Figure 3.

### ISSUE THREAD

**Issue by shangqing-liu** [tech-srl/bottleneck#7]
Hello,

Thanks for this great repo. I cannot reproduce the exact results with the default configurations as follows:
```
python run-gcn-2-8.py
python run-gat-2-8.py
python run-ggnn-2-8.py
python run-gin-2-8.py
```

The results produced by these scripts as as follows:

|      | 2 |   3  |   4  |   5  |
|:----:|:-:|:----:|:----:|:----:|
| GGNN | 1 | 0.99 | 0.93 | 0.49 |
|  GAT | 1 |   1  | 0.63 |   *  |
| GIN  | 1 | 0.99 | *    | *    |
| GCN  | 1 | 1    | 0.58 | *    |

Marker * denotes the experiments are still running.

May I ask the hyper-parameters in the default configurations are the same as the paper used？ 

Thanks.

**Comment by urialon** [comment]
Hi @shangqing-liu ,
Thank you for your interest in our paper!

Yes, the hyper-parameters in this repository are the same as the paper used. 

The only possible difference is in the random seeds. We noticed that sometimes, a seed may get "stuck" and cannot continue improving its training accuracy. So, in such cases, we tried a different random seed and reported the higher result of the two, since our goal was to give each GNN the highest chances possible.

As it looks. your results are pretty close to our results, except maybe for GAT at `depth=4` which *can* reach 100%.
This is not one of these datasets where every minor improvement is important and an improvement of 1% matters. The main point here was to show that the results in general are far from 100%, especially as depth increases. In particular, at a depth of 4 there is a difference between {GCN,GIN} vs. {GAT, GGNN} that we discuss in the paper.

Let me know if you have any more questions!
Uri

**Comment by shangqing-liu** [comment]
Hi Uri,

The

**Comment by shangqing-liu** [comment]
Hi Uri,

Thanks for the responses and I will try different seeds. By the way, the paper is really interesting and inspiring and I will try some follow-up works based on it.

Thanks very much,

Shangqing

**Comment by urialon** [comment]
Thank you for your kind words!
